sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -37,6 +37,9 @@ private[sql] case class JDBCPartitioningInfo(
     numPartitions: Int)
 
 private[sql] object JDBCRelation {
+
+  import scala.collection.JavaConverters._
```

Comment:
Hey @tribbloid, #10519 has now been merged, so I think things should be in a much more testable state now in case you want to revive this PR and try to get it merged.